### ISSUE TITLE
No error if abaplint Instance not available

### ISSUE THREAD

**Issue by BlackmanCC** [abaplint/abaplint-sci-client#275]
abaplint-sci-client does not receive an error if the abaplint instance is  not available (e.g. docker container not running).

ZCL_ABAPLINT_BACKEND_API_AGENTCP->PARSE_RESPONSE gets lv_code "502" and lv_sreason "Bad Gateway" and runs into 
zcx_abapgit_ajson_error with message "JSON PARSER: Node without name (maybe not JSON) @ /"

The exception is caught in ZCL_ABAPLINT_CHECK->GET_MAP, but not handled.

**Comment by BlackmanCC** [comment]
I think the code in ZCL_ABAPLINT_BACKEND_API_AGENTCP->PARSE_RESPONSE should be changed like this:
before:
```
    IF lv_response IS INITIAL.
      RAISE EXCEPTION TYPE zcx_abaplint_error
        EXPORTING
          message = |API request failed [{ lv_scode }]|.
    ENDIF.
```

after:
```
    IF lv_scode <> IF_HTTP_STATUS=>REASON_200.
      RAISE EXCEPTION TYPE zcx_abaplint_error
        EXPORTING
          message = |API request failed [{ lv_scode }]|.
    ENDIF.
```
